docs/design/unsupported-args-design.md

```diff
@@ -0,0 +1,118 @@
+   
+   This approach also introduces more complexity as it requires logic to handle and validate each of the options. On top of that having CRD options introduces migration and upgrade problems requiring migration strategies to be taken into account for existing resources. 
+
+   Adding the unsupported arguments and options within CRD makes those options highly visible to the users and may encourage to use them.
+
+ - Annotations to include unsupported args directly
+   This mmethod allows easily to add args to existing resources, however they have size limitations, restricting the amount of data that can be stored and may have impact on the scenarios where limit is reached.
+
+   Annotations are also problematic to maintain when used for more complex types of configurations.
+
+   Sample custom unsupported args:
+
+    ```yaml
+    kind: DataProtectionApplication
+    apiVersion: oadp.openshift.io/v1alpha1
+    metadata:
+    name: sample-dpa
+    namespace: openshift-adp
+    annotations:
+        custom.configmap.dpa/velero-server-args: '{"arg_1":"val_1","arg2":"val2"}'
+        custom.configmap.dpa/node-agent-args: '{"arg_1":"val_1","arg_2":"val_2"}'
+    spec:
+        [...]
+    ```
+
+ - Labels to include unsupported args directly
+
+   This approach is similar to annotations, with all the pros/cons of them with additional cons of labels being designed for resource identification, querying, and organizing within clusters and not configurations.
+
+## Security Considerations
+Passing input data without validation may cause malformed configurations to be included.
+
+This design does not allow to store sensitive information within configuration, as it's using ConfigMaps and not Secrets.
+
+Proper RBAC access restriction to the relevant ConfigMaps is required to ensure least privilege is granted.
+
+## Compatibility
+This design is backwards compatible with previous unsupported arguments passed via DPA specification, however it takes precendence over them.
+
+This may lead to the situation where users will have running previous configurations that are not reflected with the OADP instances that are using new way of defining unsupported arguments.
+
+## Implementation
+The implementation will take place within OADP controller only, without changes to the DPA specification.
+
+Once ConfigMap(s) are discovered to be in the same Namespace as OADP their config will take precendence and replace all the other arguments passed to the executable as a simple string without any validations.
+
+## Open Issues
+ - Should we include the unsupported args in all cases when the ConfigMaps are found in the napespace or maybe only when annotations are within DPA?
+ - Naming convention for the annotations - currently it's not well defined
+ - We may consider using one ConfigMap instead of multiple ConfigMaps, but this will lock down it's schema and may lead to future issues with enabling other unsupported args.
```

Comment:
I believe we should use 2 configmaps - one for velero and the other for node-agent, and these should have 2 separate annotation keys too.

Reply:
Removed from Open Issues and added to the examples.